### PR TITLE
Add jest config and setup

### PR DESCRIPTION
**Pull request by Shobhit1**
- This PR will add initial config and setup for jest.
- adds jest config
- adds a placeholder typescript test
- adds a watch step and a normal `test` step in `package.json`
- makes sure that coverage is being generated & reported
- creates a `test-results` folder to store those tests

### CODE REVIEW

File: tests/unit/jest.config.js [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+    rootDir: rootDir,
+    roots: [currentDir],
+    collectCoverage: true,
+    collectCoverageFrom: [
+        '<rootDir>/src/**/*.{ts,tsx}',
+        '!<rootDir>/tests/**/*',
+        '!<rootDir>/src/**/*.d.ts',
+    ],
+    coverageReporters: ['json', 'lcov', 'text', 'cobertura'],
+    testEnvironment: 'jsdom',
```
Comment (dbjorge) [inline tests/unit/jest.config.js]: Do we need this? Does it even work without a dependency on the jsdom package?

File: tests/unit/jest.config.js [hunk 2]
```diff
@@ -0,0 +1,37 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+const rootDir = '../../';
+const currentDir = '<rootDir>/tests/unit';
+
+module.exports = {
+    clearMocks: true,
+    testRunner: 'jest-circus/runner',
+    transform: {
+        '^.+\\.(ts|tsx)$': 'ts-jest',
```
Comment (dbjorge) [inline tests/unit/jest.config.js]: I don't this this project should have tsx files; does our jest config need to support them? (same for lines 15, 20, 26)

File: tests/unit/jest.config.js [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+    coverageDirectory: '<rootDir>/test-results/unit/coverage',
+    displayName: 'unit tests',
+    moduleFileExtensions: ['ts', 'tsx', 'js'],
+    rootDir: rootDir,
+    roots: [currentDir],
+    collectCoverage: true,
+    collectCoverageFrom: [
+        '<rootDir>/src/**/*.{ts,tsx}',
+        '!<rootDir>/tests/**/*',
+        '!<rootDir>/src/**/*.d.ts',
```
Comment (dbjorge) [inline tests/unit/jest.config.js]: I don't think we should have any .d.ts files in /src/, so I feel inclined to not configure coverage for them

File: .gitignore [hunk 4]
```diff
@@ -59,3 +59,6 @@ typings/
 
 # next.js build output
 .next
+
+# test-results needed for CI but not needed on remote
```
Comment (dbjorge) [inline .gitignore]: This comment confuses me, and I don't think it's necessary; I think it's going to be apparent to a reader why a folder named "test-results" should be .gitignored :)

**Comment by Shobhit1** [comment]
Got it. I tried to not include a lot of React stuff, guess some got left. I will get rid of it asap.

File: package.json [hunk 5]
```diff
@@ -5,12 +5,19 @@
...
+        "@types/jest": "^24.0.11",
+        "jest": "^24.5.0",
+        "jest-circus": "^24.5.0",
+        "jest-junit": "^6.3.0",
         "prettier": "^1.16.4",
+        "ts-jest": "^24.0.1",
         "tslint-microsoft-contrib": "^6.1.0"
     },
     "scripts": {
+        "test": "jest",
```
Comment (dbjorge) [inline package.json]: >"test": "jest", [](start = 8, length = 15)

nitpick: can we keep these alphabetized like npm does for the dependencies?

**Pull request merged**